Jenkinsfile

```diff
@@ -0,0 +1,162 @@
+ *     developed by Intel Corporation and credit the contributors.
+ *
+ * 5. Neither the name of Intel Corporation, nor the name of any Contributor
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+// To use a test branch (i.e. PR) until it lands to master
+// I.e. for testing library changes
+//@Library(value="pipeline-lib@your_branch") _
+
+pipeline {
+    agent none
+
+    stages {
+        stage('Cancel Previous Builds') {
+            when { changeRequest() }
+            steps {
+                cancelPreviousBuilds()
+            }
+        }
+        stage('Lint') {
+            stages {
+                stage('RPM Lint') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos.7'
+                            label 'docker_runner'
+                            additionalBuildArgs  '--build-arg UID=$(id -u)'
+                            args  '--group-add mock --cap-add=SYS_ADMIN --privileged=true'
+                        }
+                    }
+                    steps {
+                        // slurm.spec fails lint.
+                        sh script: '''make spec
+                                      make rpmlint''',
+                           returnStatus: true
+                    }
+                }
+            }
+        }
+        stage('Build') {
+            parallel {
+                stage('Build on CentOS 7') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.centos.7'
+                            label 'docker_runner'
+                            additionalBuildArgs '--build-arg UID=$(id -u)'
+                        }
+                    }
+                    steps {
+                        sh '''rm -rf artifacts/centos7/
+                              mkdir -p artifacts/centos7/
+                              make spec
+                              make srpm
+                              ln _topdir/SRPMS/* artifacts/centos7/
+                              make rpms
```

Comment:
If `make rpms` works, `make mockbuild` should work, and if it doesn't it's doing it's job and pointing out a dirty specfile.

What are the errors/failures you get, specifically?

Reply:
Found the cause of the issue, needed to pass a build option to mock for it to pull in all the libraries needed for the build.  Probably a bug in the slurm.spec file where it is always expecting to build a package documented as optional.

Reply:
Yeah.  A missing `--with` option should not produce build failures.  Sounds like a bug in the specfile.